### PR TITLE
Refactors and Improvised Heating

### PR DESCRIPTION
**Pull request by TheDZD**
- Refactors the is_hot proc, is now nearly the same as TG's.
- Refactors a bunch of stuff in food/drinks.dm and
reagent_containers/glass.dm, screw you relative pathing.
- Allows you to heat reagent containers with anything that is_hot.
- Possibly invokes an [ancient Mayan curse](http://i.gyazo.com/2b5fa7db996f590dab0ce905861930f5.png) in reagent_containers/glass.dm

DNM until I test and figure out what the hell Git was doing with glass.dm.

### CODE REVIEW

**Comment by TheDZD** [comment]
What the actual hell Git? File changes for glass.dm look really odd.

**Comment by Fox-McCloud** [comment]
yeah, and you're running into the same issue I did; it still thinks that is_hot and +temp to beakers is already a thing.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -290,14 +313,15 @@
 	volume = 120
 	flags = OPENCONTAINER
```
Comment (TheDZD) [inline code/modules/reagents/reagent_containers/glass.dm]: Compare to [here](https://github.com/ParadiseSS13/Paradise/pull/783), this was not here prior to this PR.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -290,14 +313,15 @@
...
-
			qdel(src)
+
/obj/item/weapon/reagent_containers/glass/bucket/attackby(var/obj/D, mob/user as mob, params)
+
	if(isprox(D))
+
		user << "You add [D] to [src]."
+
		del(D)
```
Comment (TheDZD) [inline code/modules/reagents/reagent_containers/glass.dm]: Yet somehow it registered this as being new.

**Comment by TheDZD** [comment]
Whoops.
Just wanted to add, all these extra line breaks were not manually added by me, but by Git. 
/play trombone

**Comment by TheDZD** [comment]
See #784

**Pull request closed without merging**